### PR TITLE
fix: add field sync_product_id to product

### PR DESCRIPTION
**Pull request by svengau**
https://github.com/ynnoj/gatsby-source-printful/issues/15

### CODE REVIEW

**Comment by svengau** [comment]
Hi @ynnoj, there are a couple of bugs in v2.0.0-beta.1 regarding this fix. Sorry for that, I've been too quick proposing a PR. 

In fact, we can't use an integer to represent the id in Gatsby (and printful id is an integer). 

I've also noticed product id = variant -> sync_product_id, so instead of introducing a breaking change, I've added a new field product -> sync_product_id.

**Comment by ynnoj** [comment]
@svengau I think it's best the library mimic the Printful API as close as possible.

Your original PR was the right way to go, we just need to covert those pesky ID integer fields to strings.

**Comment by svengau** [comment]
@ynnoj thanks for the feedback, I will update the PR accordingly.

**Comment by svengau** [comment]
Hi @ynnoj, I hope this one is the good one :) 
I've tested it on my project, with shipping rates, and order creation, and it performs well. 

I've named the local variables sync_product_id  for product.id and sync_variant_id for variant.id, to make it clearer. But of course at the end, the exact same data structure than printful is exposed.

File: gatsby-source-printful/gatsby-node.js [hunk 1]
```diff
@@ -73,14 +73,14 @@ exports.sourceNodes = async (
...
-    const { external_id, variants: variantCount, ...rest } = product
+    const { variants: variantCount, id: sync_product_id, ...rest } = product
 
     let productImageNode
 
     try {
       const { id } = await createRemoteFileNode({
         url: product.thumbnail_url,
-        parentNodeId: external_id,
+        parentNodeId: sync_product_id.toString(),
```
Comment (ynnoj) [inline gatsby-source-printful/gatsby-node.js:83]: @svengau Let's declare a constant here as this is used in a couple spots.

File: gatsby-source-printful/gatsby-node.js [hunk 2]
```diff
@@ -117,7 +118,7 @@ exports.sourceNodes = async (
       try {
         const { id } = await createRemoteFileNode({
           url: previewFile.preview_url,
-          parentNodeId: external_id,
+          parentNodeId: sync_variant_id.toString(),
```
Comment (ynnoj) [inline gatsby-source-printful/gatsby-node.js:121]: @svengau Let's declare a constant here as this is used in a couple spots.

File: gatsby-source-printful/gatsby-node.js [hunk 3]
```diff
@@ -108,7 +108,8 @@ exports.sourceNodes = async (
   }
 
   const processVariant = async ({ variant, product }) => {
-    const { external_id, variant_id, ...rest } = variant
+    const { id: sync_product_id } = product
```
Comment (ynnoj) [inline gatsby-source-printful/gatsby-node.js:111]: @svengau No need to destructure.

File: gatsby-source-printful/gatsby-node.js [hunk 4]
```diff
@@ -132,11 +133,11 @@ exports.sourceNodes = async (
 
     const nodeData = {
       ...rest,
+      id: sync_variant_id.toString(),
       slug: parseNameForSlug(variant.name),
       retail_price: parsePriceString(variant.retail_price),
-      parentProduct___NODE: product.external_id,
+      parentProduct___NODE: sync_product_id.toString(),
```
Comment (ynnoj) [inline gatsby-source-printful/gatsby-node.js:139]:
```suggestion
      parentProduct___NODE: product.id.toString(),
```

**Comment by svengau** [comment]
hey @ynnoj, 

I've rebased the PR on beta, and completed the requested changes.

**Comment by ynnoj** [comment]
@svengau Try out that new release. If all good, will merge to `master`.

**Pull request merged**